src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -816,17 +874,33 @@ protected override void PopulateEncLogTableRows(ImmutableArray<int> rowCounts)
         private void PopulateEncLogTableParameters()
         {
             var parameterFirstId = _parameterDefs.FirstRowId;
-            for (int i = 0; i < _parameterDefList.Count; i++)
+            int i = 0;
+            foreach (var paramDef in GetParameterDefs())
             {
-                var methodDef = _parameterDefList[i].Key;
+                var methodDef = _parameterDefList[paramDef];
 
-                metadata.AddEncLogEntry(
-                    entity: MetadataTokens.MethodDefinitionHandle(_methodDefs[methodDef]),
-                    code: EditAndContinueOperation.AddParameter);
+                if (_changes.AlwaysEmitParams(methodDef.GetInternalSymbol()))
+                {
+                    // For previously emitted parameters we just update the Param row
+                    var param = GetParameterHandle(paramDef);
+                    _paramEncMapRows.Add(MetadataTokens.GetRowNumber(param));
+                    metadata.AddEncLogEntry(
+                        entity: param,
+                        code: EditAndContinueOperation.Default);
+                }
+                else
+                {
+                    // For paramteres on new methods we emit AddParameter rows for the method too
+                    _paramEncMapRows.Add(parameterFirstId + i);
+                    metadata.AddEncLogEntry(
+                        entity: MetadataTokens.MethodDefinitionHandle(_methodDefs[methodDef]),
+                        code: EditAndContinueOperation.AddParameter);
 
-                metadata.AddEncLogEntry(
-                    entity: MetadataTokens.ParameterHandle(parameterFirstId + i),
-                    code: EditAndContinueOperation.Default);
+                    metadata.AddEncLogEntry(
+                        entity: MetadataTokens.ParameterHandle(parameterFirstId + i),
```

Comment:
>parameterFirstId + i [](http://example.com/codeflow?start=63&length=20)

If an edit has a mix of methods, some with parameters that are always emitted and some without (new methods?), is there a chance the parameter handles collide?